optimade/filtertransformers/mongo.py

```diff
@@ -86,3 +86,180 @@ def combined(self, args):
                     val = val[1:-1]
             elements.append(val)
         return elements
+
+
+class NewMongoTransformer(Transformer):
+    operator_map = {"<": "$lt", "<=": "$lte", ">": "$gt", ">=": "$gte", "!=": "$ne", "=": "$eq"}
+    reversed_operator_map = {"$lt": "$gt", "$lte": "$gte", "$gt": "$lt", "$gte": "$lte", "$ne": "$ne", "$eq": "$eq"}
+
+    def __init__(self):
+        super().__init__()
+
+    def filter(self, arg):
+        # filter: expression*
+        if not arg:
+            return None
+        return arg[0]
+
+    @v_args(inline=True)
+    def constant(self, value):
+        # constant: string | number
+        # Note: Do nothing!
+        return value
+
+    @v_args(inline=True)
+    def value(self, value):
+        # value: string | number | property
+        # Note: Do nothing!
+        return value
+
+    def value_list(self, arg):
+        # value_list: [ OPERATOR ] value ( "," [ OPERATOR ] value )*
+        raise NotImplementedError
+
+    def value_zip(self, arg):
+        # value_zip: [ OPERATOR ] value ":" [ OPERATOR ] value (":" [ OPERATOR ] value)*
+        raise NotImplementedError
+
+    def value_zip_list(self, arg):
+        # value_zip_list: value_zip ( "," value_zip )*
+        raise NotImplementedError
+
+    def expression(self, arg):
+        # expression: expression_clause ( OR expression_clause )
+        # expression with and without 'OR'
+        return {'$or': arg} if len(arg) > 1 else arg[0]
+
+    def expression_clause(self, arg):
+        # expression_clause: expression_phrase ( AND expression_phrase )*
+        # expression_clause with and without 'AND'
+        return {'$and': arg} if len(arg) > 1 else arg[0]
+
+    def expression_phrase(self, arg):
+        # expression_phrase: [ NOT ] ( comparison | predicate_comparison | "(" expression ")" )
+        if len(arg) == 1:
+            # without NOT
+            return arg[0]
+        else:
+            # with NOT
+            # TODO: This implementation probably fails in the case of `predicate_comparison` or `"(" expression ")"`
```

Comment:
These things are meant to be fixed/handled in another PR, correct? Because then I will ignore it here :)

Reply:
This one is tricky and it needs its own issue (#79). There is an example of this issue:
```
"NOT a>1" -> {'a': {'$not': {'$gt': 1}}}
"NOT a>1 AND b>1" -> {'$and': [{'a': {'$not': {'$gt': 1}}}, {'b': {'$gt': 1}}]}
"NOT (a>1 AND b>1)" -> {'$and': {'$not': [{'a': {'$gt': 1}}, {'b': {'$gt': 1}}]}}
```
In the case of the last one the transformer gives you a rubish query. Previously this issue was there but it was completly ignored...